apps/src/templates/textResponses/TextResponses.jsx

```diff
@@ -23,150 +24,120 @@ const CSV_HEADERS = [
   {label: i18n.question(), key: 'question'},
   {label: i18n.response(), key: 'response'}
 ];
-const DEFAULT_FILTER_KEY = i18n.all();
 const PADDING = 8;
 
-class TextResponses extends Component {
-  static propTypes = {
-    // Provided by redux.
-    sectionId: PropTypes.number.isRequired,
-    responses: PropTypes.object.isRequired,
-    isLoadingResponses: PropTypes.bool.isRequired,
-    validScripts: PropTypes.arrayOf(validScriptPropType).isRequired,
-    scriptId: PropTypes.number,
-    scriptName: PropTypes.string,
-    setScriptId: PropTypes.func.isRequired,
-    asyncLoadTextResponses: PropTypes.func.isRequired
-  };
-
-  state = {
-    filterByLessonName: null
-  };
-
-  componentDidMount() {
-    this.props.asyncLoadTextResponses(
-      this.props.sectionId,
-      this.props.scriptId
-    );
-  }
-
-  getResponsesByScript = () => {
-    const {responses, scriptId} = this.props;
-    return responses[scriptId] || [];
-  };
-
-  onChangeScript = scriptId => {
-    const {setScriptId, asyncLoadTextResponses, sectionId} = this.props;
-    asyncLoadTextResponses(sectionId, scriptId, () => {
-      setScriptId(scriptId);
-      this.setState({filterByLessonName: null});
-    });
-  };
-
-  renderFilterByLessonDropdown = () => {
-    const lessons = this.getLessons();
-
-    // only render filter dropdown if there are 2+ lessons
-    if (lessons.length <= 1) {
-      return null;
+function TextResponses({
+  sectionId,
+  validScripts,
+  scriptId,
+  scriptName,
+  setScriptId
+}) {
+  const [textResponsesByScript, setTextResponsesByScript] = useState({});
+  const [isLoadingResponses, setIsLoadingResponses] = useState(false);
+  const [filterByLessonName, setFilterByLessonName] = useState(null);
+
+  useEffect(() => {
+    asyncLoadTextResponses(sectionId, scriptId);
+  }, [scriptId]);
```

Comment:
If sectionId is an input to `asyncLoadTextResponses`, do we need to include in the `deps` parameter to `useEffect`?

Reply:
Yes! I'll add it, the reason that I didn't was because I happen to know that sectionId isn't going to change over the lifetime of the component, but that could change in the future